### PR TITLE
Add device endpoints to push/pull files and pull directory.

### PR DESCRIPTION
**Pull request by bmarroquin**
implementation of endpoints can be found here: https://github.com/appium/appium/blob/e8886ae0d276ac4f97a1bdb88998fc30fbd936fa/packages/base-driver/lib/protocol/routes.js#L648

They don't have modern documentation for this but there is some old documentation found here:
https://appium.readthedocs.io/en/latest/en/commands/device/files/push-file/
https://appium.readthedocs.io/en/latest/en/commands/device/files/pull-file/
https://appium.readthedocs.io/en/latest/en/commands/device/files/pull-folder/

### CODE REVIEW

**Comment by aristotelos** [comment]
I am wondering whether this should be implemented in FlaUI.WebDriver directly, or instead in an NPM wrapper library that provides it as Appium driver. For example, https://github.com/appium/appium-windows-driver/blob/master/lib/commands/file-movement.js implements this in the Appium driver, not in the `WinAppDriver`. Is there a reason to implement this and other Appium commands in FlaUI.WebDriver directly?

**Comment by aristotelos** [comment]
If we decide to implement this functionality here (see previous comment), the README should be extended as well.

**Review by aristotelos — CHANGES_REQUESTED** [review]
Thank you for your contribution. I have added some questions and minor suggestions.

File: src/FlaUI.WebDriver/Controllers/DeviceController.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                // data could be an empty string, so just a simple null check
+                return MissingParameter("data");
+            }
+            if (string.IsNullOrEmpty(request.Path))
+            {
+                return MissingParameter("path");
+            }
+            var parent = Path.GetDirectoryName(request.Path)!;
+            if (!Directory.Exists(parent))
```
Comment (aristotelos) [inline src/FlaUI.WebDriver/Controllers/DeviceController.cs:50]: To add a little bit of security, we could use a "feature flag" mechanism such as appium-windows-driver to only allow this endpoint if it has been enabled, see https://github.com/appium/appium-windows-driver/blob/master/lib/commands/file-movement.js#L18

**Comment by bmarroquin** [comment]
> I am wondering whether this should be implemented in FlaUI.WebDriver directly, or instead in an NPM wrapper library that provides it as Appium driver. For example, https://github.com/appium/appium-windows-driver/blob/master/lib/commands/file-movement.js implements this in the Appium driver, not in the `WinAppDriver`. Is there a reason to implement this and other Appium commands in FlaUI.WebDriver directly?

Interesting, i didn't realize that the Appium server layer added features. I was avoiding running the Appium wrapper because we need to jump through some hoops to get things to run in our Windows test environment, so I wanted to avoid having to configure node.js there. We get more features though so it could be worth the extra work. I will close for now.

Thanks.

**Pull request closed without merging**